float32 maxF
float32 minF
---
bool ack